ENH Add verbose option to SpectralClustering (#18052)
scikit-learn/scikit-learn: doc/whats_new/v0.24.rst

  :user:`Emilie Delattre <EmilieDel>`, and
  :user:`Francesco Casalegno <FrancescoCasalegno>`.

> - |Enhancement| :class:`cluster.SpectralClustering` and
>   :func:`cluster.spectral_clustering` have a new optional argument `verbose`.
>   When set to `True`, additional messages will be output that can aid in
>   debugging. :pr:`18052` by :user:`Sean O. Stalley <sstalley>`.

Reviewer: ```suggestion
- |Enhancement| :class:`cluster.SpectralClustering` and
  :func:`cluster.spectral_clustering` has a new keyword argument `verbose`.
  When set to `True`, additional messages will be outputted, which can aid with
  debugging. :pr:`18052` by :user:`Sean O. Stalley <sstalley>`.
```
